File: codes/day-2/payrollapp/src/models/Developer.java
package models;

public class Developer extends Employee {
    private double incentivePayment;

    public Developer(int id) {
        super(id);
    }

    public Developer(int id, String name, double incentivePayment, double basicPayment, double daPayment,
            double hraPayment) {
        super(id, name, basicPayment, daPayment, hraPayment);
        this.incentivePayment = incentivePayment;
    }

    public double getIncentivePayment() {
        return incentivePayment;
    }

    public void setIncentivePayment(double incentivePayment) {
        this.incentivePayment = incentivePayment;
    }

    @Override
    public void calculateSalary() {
        super.calculateSalary();
        double total = getTotalPayment() + this.incentivePayment;
        setTotalPayment(total);
    }
}

File: codes/day-12/payroll-app/src/models/developer.js
import { Employee } from "./employee.js";

export class Developer extends Employee {
  constructor(id, name, basic, da, hra, incentive) {
    super(id, name, basic, da, hra);
    this.incentivePayment = incentive;
  }
  calculateSalary() {
    super.calculateSalary();
    this.totalPayment += this.incentivePayment;
  }
}
